pages/mercury_tour_page.py

```diff
@@ -1,73 +1,134 @@
+
+    def select_departing_location(self, from_port):
+        """
+        Enter from_port from field
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.NAME, 'fromPort'))
+        )).select_by_visible_text(from_port)
+
+    def select_departing_month(self, from_month):
+        """
+        Select from_month from dropdown
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.NAME, 'fromMonth'))
+        )).select_by_visible_text(from_month)
+
+    def select_departing_day(self, from_day):
+        """
+        Select from_day from dropdown
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.NAME, 'fromDay'))
+        )).select_by_visible_text(from_day)
+
+    def select_arriving_port(self, to_port):
+        """
+        Select to_port
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.NAME, 'toPort'))
+        )).select_by_visible_text(to_port)
+
+    def select_returning_month(self, to_month):
+        """
+        Select to_month from list
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.NAME, 'toMonth'))
+        )).select_by_visible_text(to_month)
+
+    def select_returning_day(self, to_day):
+        """
+        Select to_day from list
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.NAME, 'toDay'))
+        )).select_by_visible_text(to_day)
+
+    def select_business_class(self):
+        """
+        Select Business class from dropdown
+        """
+        WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.CSS_SELECTOR, 'input[value="Business"]'))
+        ).click()
+
+    def select_an_airline(self, airline_name):
+        """
+        Choose an airline
+        """
+        Select(WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.visibility_of_element_located((By.CSS_SELECTOR, 'select[name="airline"]'))
+        )).select_by_visible_text(airline_name)
+
+    def submit_flight_details(self):
+        """
+        After entering all credentials submit form and wait to open next page
+        """
         self.driver.find_element_by_css_selector('input[name="findFlights"]').click()
-
-    def hotels_ver_click(self):
-        # Click hotels vertical
-        self.driver.find_element_by_link_text('Hotels').click()
-
-    def back_click(self):
-        self.driver.find_element_by_css_selector('img[src="images/home.gif"]').click()
+        contact = WebDriverWait(self.driver, self.time_to_wait).until(
+            EC.presence_of_element_located((By.LINK_TEXT, 'CONTACT'))
```

Comment:
the `CONTACT` tab appears on all the pages. we should find something that `uniquely` identify this page.

Reply:
As i had discussed before its really hard for this site specially to find something unique .So you said images are not reliable so then there title assert is the only option i have here.